packages/gestalt/src/Dropdown.js

```diff
@@ -95,46 +95,40 @@ export default function Dropdown({
       cursorIndex = optionsCount;
     }
 
-    if (availableOptions[cursorIndex] && availableOptions[cursorIndex].props) {
-      const newItem = availableOptions[cursorIndex].props.option;
+    const { props: cursorOption } = allowedChildrenOptions[cursorIndex];
+
+    if (cursorOption) {
+      const item = cursorOption.option;
       setHoveredItem(cursorIndex);
 
       if (direction === KEYS.ENTER) {
-        // $FlowFixMe[incompatible-call]
-        if (onSelect) onSelect({ event, item: newItem });
-        if (availableOptions[cursorIndex].props.handleSelect)
-          availableOptions[cursorIndex].props.handleSelect({
-            event,
-            item: newItem,
-          });
+        onSelect?.({ event, item });
+        cursorOption.handleSelect?.({
+          event,
+          item,
+        });
       }
     }
-
-    // Scrolling
-    handleContainerScrolling(direction, containerRef, selectedElement);
```

Comment:
I believe this was doing nothing! but please, correct me if I'm missing something. container was always undefined. Without it, still works.